### ISSUE TITLE
Flickering when modifying main area's ratio

### ISSUE THREAD

**Issue by chcore** [djpohly/dwl#306]
## Info
dwl's commit: ba7dcb2dea2d5dc46b73b6b9194daab94a325a05 (master at time of writing, but also happens with the v0.3.1 release)
wlroots version: 0.15.1
## Description

There is noticeable flickering when resizing the ratio of the main area. I'm using the default config, with no patches or modifications. Not shown in the video is the flicker overflowing onto the next display on the right, when increasing the ratio.

https://user-images.githubusercontent.com/83099052/190829920-942f20ae-b38f-4e21-92ce-4155a23ab40a.mp4

Note that the above video happens to show dwl running inside another compositor, but the exact same thing happens when running it standalone.

Thanks

**Comment by PalanixYT** [comment]
That behaviour is known and a fix is already almost finished.

**Comment by pm4rcin** [comment]
@PalanixYT any progress or a patch to test?

**Comment by PalanixYT** [comment]
Yeah you can look at v2flick branch, there is the fix. It was working great until firefox started acting up, refusing to update unless given focus (this requires you to have 2 monitors) to reproduce you spawn firefox on monitor 1 and then spawn another client, like foot, on monitor 2. then move the extra client to monitor 1. firefox will still be fine, display will update. Then, repeat this step and move another client, the display will freeze.
tldr; no idea what I'm doing, @sevz17 now does it

**Comment by PalanixYT** [comment]
Turns out it was just a problem with wlroots because merging it with wlroots-next works.

**Comment by sevz17** [comment]
Fixed in 7eaa01ac1f074511ae1013326172d51c6fdf8866

**Comment by chcore** [comment]
This is not fixed if your `repeat_rate` and `repeat_delay` are fast enough. I use these:

`static const int repeat_rate = 50;`
`static const int repeat_delay = 300;`

And am seeing the exact same issue. For comparison, Sway does not have this problem when configured with the same repeat and delay settings.

**Comment by sevz17** [comment]
> This is not fixed if your `repeat_rate` and `repeat_delay` are fast enough. I use these and am still seeing the exact same issue on master:
> 
> `static const int repeat_rate = 50;` `static const int repeat_delay = 300;`
> 
> For comparison, Sway does not have this problem when configured with the same repeat and delay settings.

Actually, this will not be 100% fixed until scene-graph implements atomic commits.

**Comment by chcore** [comment]
@sevz17 is that feature tracked anywhere?
